lib/instrumentation/elasticsearch-shared.js

```diff
@@ -26,8 +26,10 @@ exports.pathIsAQuery = pathIsAQuery
 //
 //    {}
 //    {"query":{"query_string":{"query":"pants"}}}
+//
+// `path`, `query`, and `body` can all be null or undefined.
 exports.setElasticsearchDbContext = function (span, path, query, body) {
-  if (pathIsAQuery.test(path)) {
+  if (path && pathIsAQuery.test(path)) {
```

Comment:
nit: Is the `path &&` needed here?  My naive reading is `pathIsAQuery.test(path)` should be false for all the false-ish values.

```
const pathIsAQuery = /\/(_search|_msearch|_count|_async_search|_sql|_eql)(\/|$)/

> pathIsAQuery.test(null)
false
> pathIsAQuery.test(undefined)
false
> pathIsAQuery.test('')
false
> pathIsAQuery.test(false)
false

etc.
```

Reply:
Yah, true, but passing other than a string into `RegExp.prototype.test` strikes me as messy. It is doing implicit type conversion to string and doing a regex match on that. For example:

```
> pathIsAOops = /foo|bar|null|baz$/
/foo|bar|null|baz$/
> pathIsAOops.test(null)
true
```